src/PJV.ts

```diff
@@ -212,11 +212,16 @@ type ValidationOutput = {
 	critical?: string | Record<string, string>;
 };
 const validate = (
-	data: string,
+	data: string | object,
 	specName: SpecName = "npm",
 	options: ValidationOptions = {},
 ): ValidationOutput => {
-	const parsed = parse(data);
+	let parsed;
+	if (typeof data !== "object") {
+		parsed = parse(data);
+	} else {
+		parsed = data;
+	}
```

Comment:
[Style] Nitpicking, no need for this to be a `let` (https://github.com/sindresorhus/eslint-plugin-unicorn/blob/e5e17f8edd7bee0b375313c3708955980cbbfe96/docs/rules/prefer-ternary.md):

```suggestion
	const parsed = typeof data === "object" ? data : parse(data);
```